Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -204,36 +204,37 @@ extension NavigationMapView {
     }
     
     func updateRouteLineGradientStops(fractionTraveled: Double, gradientStops: [Double: UIColor]) -> [Double: UIColor] {
-        var filteredGradientStops = gradientStops.filter { key, value in
-            return key >= fractionTraveled
+        // minimumSegment records the nearest smaller or equal stop and associated congestion color of the `fractionTraveled`, and then apply it's color to the `fractionTraveled` stop.
+        var minimumSegment: (Double, UIColor) = (0.0, trafficUnknownColor)
+        var filteredGradientStops = [Double: UIColor]()
+        
+        for (key,value) in gradientStops {
+            if key > fractionTraveled {
+                filteredGradientStops[key] = value
+            } else if key >= minimumSegment.0 {
+                minimumSegment = (key, value)
+            }
         }
-        filteredGradientStops[0.0] = traversedRouteColor
         
+        filteredGradientStops[0.0] = traversedRouteColor
         let  nextDownFractionTraveled = Double(CGFloat(fractionTraveled).nextDown)
         if nextDownFractionTraveled >= 0.0 {
             filteredGradientStops[nextDownFractionTraveled] = traversedRouteColor
         }
-        
-        // Find the nearest smaller stop than the `fractionTraveled` and apply its color to the `fractionTraveled` till the next stop.
-        let sortedStops = gradientStops.keys.sorted()
-        if let minStop = sortedStops.last(where: { $0 <= fractionTraveled }),
-           minStop != 0.0 {
-            filteredGradientStops[fractionTraveled] = gradientStops[minStop]
-        } else {
-            filteredGradientStops[fractionTraveled] = trafficUnknownColor
-        }
+        filteredGradientStops[fractionTraveled] = minimumSegment.1
 
         return filteredGradientStops
     }
     
-    func routeLineGradient(_ congestionFeatures: [Turf.Feature]? = nil, fractionTraveled: Double, isMain: Bool = true) -> [Double: UIColor] {
+    func routeLineGradient(_ congestionFeatures: [Turf.Feature]? = nil, fractionTraveled: Double, isMain: Bool = true, isSoft: Bool = false) -> [Double: UIColor] {
         var gradientStops = [Double: UIColor]()
-        var distanceTraveled = fractionTraveled
+        var distanceTraveled = 0.0
```

Comment:
Is this correct change? What about re-using `fractionTraveled`?

Reply:
Yes, I think it's a correct change.

Because the `distanceTraveled ` is used to record the distance on the route in meters, and then we use the percentage of the `distanceTraveled / routeDistance` to get the traveled segment start or end percentage to compare with `fractionTraveled`. So it should start from `0.0` meters. 

Because the `fractionTraveled` is the percentage between `0` and `1`, while the `distanceTraveled` is distance in meters, previously the small fraction number doesn't affect much in the distance in meters. But it's not correct, especially if the distance of the whole route or the first step is less than 1 meter.